src/index.ts

```diff
@@ -119,34 +119,43 @@ const index = ({
   profile.originOffset = [...machine.originOffset, 0];
 
   // bed offset Z logic:
-  // - ignore inputs we know we won't be using
-  // - only look at project's z-offset setting if no inputs have a material override
-  // - use the highest z-offset seen (including 0)
-  let useZOffsetFromStyle = true;
-  for (let i = 0; i < extruderCount; i++) {
-    if (drivesUsed[i]) {
-      const material = materials[i]!;
-      if (material.style.useZOffset) {
-        useZOffsetFromStyle = false;
-        break;
+  if (palette) {
```

Comment:
Can you add a comment in this `if` block explaining why we use different logic if `palette` is present?